### PR TITLE
Stop using 's' as indicator that a method returns a list

### LINKED ISSUE #5219: [Task] Stop using `s` as an indicator that we are returning a list
For example:

```go
	DescribeNodeGroupStack(ctx context.Context, nodeGroupName string) (*Stack, error)
	DescribeNodeGroupStacks(ctx context.Context) ([]*Stack, error)
```

This is super annoying and actually caused a bug to happen: https://github.com/weaveworks/eksctl/issues/5217

It's because our unit tests were mocking the wrong function since the author missed the `s`:

```go
fakeStackManager.DescribeNodeGroupStackReturns(nil, fmt.Errorf("error describing cloudformation stack"))
```

This should have been

```
fakeStackManager.DescribeNodeGroupStacksReturns(nil, fmt.Errorf("error describing cloudformation stack"))
```

It would have been more visible if the extension would say `List` instead of an additional `s`.

The test is passing because the mock was happily providing a canned `nil, nil` as a return.

This also suggests that we should more often verify if the mock was called with the right parameters.

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

We are having a couple of methods that return either some instance or a list of those instances, which are very similar in name. In fact, the one that returns a list ends in an additional `s` (e.g. `DescribeNodeGroupStack` and `DescribeNodeGroupStacks`). This means that using and testing these functions can get error prone in particular scenarios (closes #5219). We've decided to replace the `s` with `List` for such cases. 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/cfn/manager/nodegroup.go [hunk 1]
```diff
@@ -103,8 +103,8 @@ func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(ctx context.Con
 }
 
 // DescribeNodeGroupStacks calls DescribeStacks and filters out nodegroups
```
Comment (Himangini) [inline pkg/cfn/manager/nodegroup.go:105]:
```suggestion
// DescribeNodeGroupStackList calls DescribeStackList and filters out nodegroups
```
💡

File: pkg/cfn/manager/cluster.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func (c *StackCollection) createClusterTask(ctx context.Context, errs chan error
 
 // DescribeClusterStack calls DescribeStacks and filters out cluster stack
```
Comment (Himangini) [inline pkg/cfn/manager/cluster.go:44]:
```suggestion
// DescribeClusterStack calls DescribeStackList and filters out cluster stack
```

File: pkg/cfn/manager/api.go [hunk 3]
```diff
@@ -619,7 +619,7 @@ func fmtStacksRegexForCluster(name string) string {
 }
 
 // DescribeStacks describes the existing stacks
```
Comment (Himangini) [inline pkg/cfn/manager/api.go:621]:
```suggestion
// DescribeStackList describes the existing stacks
```

**Review by Himangini — COMMENTED** [review]
small nits :D

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻

**Review by Himangini — APPROVED** [review]
👍🏻

**Review by cPu1 — CHANGES_REQUESTED** [review]
`DescribeStackList` does not read well and implies that `StackList` is the resource that needs to be described, but there is no such resource. `ListStacks` is more descriptive IMO, but there's already a method by that name so we should rename the existing `ListStacks` method to something else, e.g., `ListStacksWithStatuses`.

File: pkg/ctl/cmdutils/filter/nodegroup_filter.go [hunk 4]
```diff
@@ -93,7 +94,8 @@ func (f *NodeGroupFilter) SetOnlyLocal(ctx context.Context, eksAPI awsapi.EKS, l
 
 // SetOnlyRemote uses StackLister to list existing nodegroup stacks and configures
 // the filter to exclude nodegroups already defined in the config file. It will include the
-//  nodegroups that exist in the cluster but not in the config
+//
```
Comment (cPu1) [inline pkg/ctl/cmdutils/filter/nodegroup_filter.go:97]: Redundant line.

**Review by cPu1 — APPROVED** [review]

**Pull request merged**